pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -0,0 +1,196 @@
+        dataloader = dataloader.per_device_loader(device)
+        return dataloader
+
+    def configure_ddp(self) -> None:
+        pass
+
+    def init_ddp_connection(self, global_rank: int, world_size: int) -> None:
+        pass
+
+    def set_world_ranks(self, process_idx: int) -> None:
+        self.tpu_local_core_rank = xm.get_local_ordinal()
+        self.tpu_global_core_rank = xm.get_ordinal()
+        self.global_rank = self.tpu_local_core_rank
+        self.world_size = self.num_nodes * self.num_processes
+
+    def new_process(self, process_idx: int, trainer) -> None:
+        seed = os.environ.get("PL_GLOBAL_SEED")
+        if seed is not None:
+            seed_everything(int(seed))
+
+        self.set_world_ranks(process_idx)
+
+        # set warning rank
+        rank_zero_only.rank = self.global_rank
+
+        if self.tpu_global_core_rank != 0 and trainer.progress_bar_callback is not None:
+            trainer.progress_bar_callback.disable()
+
+        self.model_to_device()
+        self.barrier()
+
+        if trainer.testing:
+            results = trainer.run_test()
+        else:
+            results = trainer.train()
+
+        self.__save_end_of_training_weights(self.lightning_module)
+        self.transfer_distrib_spawn_state_on_fit_end(results)
+
+    def __save_end_of_training_weights(self, model: LightningModule, trainer) -> None:
+        # when training ends on these platforms dump weights to get out of the main process
+        if on_colab_kaggle():
+            rank_zero_warn("cleaning up... please do not interrupt")
+            self.save_spawn_weights(model)
+
+    def model_to_device(self) -> None:
+        pass
+
+    def barrier(self, name: Optional[str] = None) -> None:
+        rendezvous(f"pl.Trainer.{name}")
+
+    def on_save(self, checkpoint: dict) -> dict:
+        """
+        Move XLA tensors to CPU before saving
+        Recommended on XLA Guide:
+        https://github.com/pytorch/xla/blob/master/API_GUIDE.md#saving-and-loading-xla-tensors
+        """
+        return move_data_to_device(checkpoint, torch.device("cpu"))
+
+    def broadcast(self, obj: object, src: int = 0) -> object:
+        buffer = io.BytesIO()
+        torch.save(obj, buffer)
+        data = bytearray(buffer.getbuffer())
+        data_tensor = torch.tensor(data).to(xm.xla_device(), dtype=torch.float)
+        data = xm.all_gather(data_tensor)
+        buffer = io.BytesIO(data.cpu().byte().numpy())
+        obj = torch.load(buffer)
+        return obj
+
+    def load_spawn_weights(self, original_model: LightningModule) -> LightningModule:
+        """
+        Load the temp weights saved in the process
+        To recover the trained model from the ddp process we load the saved weights
+        """
+
+        loaded_model = original_model
+
+        if self.is_global_zero:
+            # load weights saved in ddp
+            path = os.path.join(original_model.trainer.default_root_dir, "__temp_weight_distributed_end.ckpt")
```

Comment:
I feel like `__temp_weight_distributed_end.ckpt` should be a GLOBAL property.

Reply:
what do you mean by global property?

Reply:
```
class

    TEMPORARY_WEIGHT_PATH = "__temp_weight_distributed_end.ckpt"
    def __init__

        
        ...

    def 

        path = os.path.join(original_model.trainer.default_root_dir, self.TEMPORARY_WEIGHT_PATH)

```